etna/core/tests/test_middleware.py

```diff
@@ -0,0 +1,76 @@
+from unittest.mock import patch
+
+from django.contrib.auth import get_user_model
+from django.test import TestCase, override_settings
+
+from wagtail.core.models import Group
+
+import responses
+
+from ...ciim.tests.factories import create_record, create_response
+from ...ciim.utils import prevent_request_warnings
+
+User = get_user_model()
+
+
+@override_settings(
+    MAINTENANCE_MODE=True,
+    MAINTENENCE_MODE_ENDS="2011-11-04T00:05:23+04:00",
+)
+class TestMaintenanceMode(TestCase):
+    @responses.activate
+    @prevent_request_warnings
+    def test_503_for_record_detail(self):
+        response = self.client.get("/catalogue/id/C123456/")
+        self.assertEquals(response.status_code, 503)
+        self.assertEquals(
+            response.headers["Retry-After"], "Fri, 04 Nov 2011 00:05:23 GMT"
+        )
+
+    @prevent_request_warnings
+    def test_503_maintenance_override_for_home_page(self):
+        response = self.client.get("/")
+        self.assertEquals(response.status_code, 503)
```

Comment:
How come the `@prevent_request_warnings` decorator is needed here? I'm not sure I follow.